lib/Server.js

```diff
@@ -628,12 +632,28 @@ class Server {
         }
 
         if (typeof client.overlay !== "undefined") {
-          searchParams.set(
-            "overlay",
+          /**
+           *
+           * @param {OverlayMessageOptions} [setting]
+           * @returns
+           */
+          const encodeOverlaySettings = (setting) =>
+            typeof setting === "function"
+              ? encodeURIComponent(setting.toString())
```

Comment:
Usage of encodeURIComponent here is intentional, because without it `URLSearchParams` will encode white space into `+` character, which could not be decoded correctly in `parseURL` in client.

Alternative is to use `URLSearchParams` in client side, but I think the drawback of that is cannot support newer browser.